### DISCUSSION TITLE
tutorial_yarp-ports problem

### DISCUSSION THREAD

**Discussion by accateo** [robotology/community#199] [Q&A] [answered]
Hi,
I'm following the tutorial_yarp-ports:

https://github.com/vvv-school/tutorial_yarp-ports

I have compiled the "port folder" code and the "buffered port" code. After that, I have run the two executables and the yarp server works correctly.
I have also connect the ports:

`$ yarp connect /w /relay/in`
`$ yarp connect /relay/out /r `

but when I try to write in the "yarp write" terminal, I don't see the command sent in the "yarp read" terminal.
I have a doubt. I did everything correctly? Or I have to edit one of the two "main.cpp" and add to it a second port?

Thanks

**Comment by pattacini** [comment]
Does anyone from @robotology/ifi-younglings want to take on this?

@Iaxama @Arya07 @gmmarconi ?

**Comment by Arya07** [comment]
Hi @accateo,
yes, to complete the tutorial, you have to edit the cpp files, following the commented hints.

**Comment by pattacini** [comment, marked as answer]
The complete tutorial is contained in the branch [`solution`](https://github.com/vvv-school/tutorial_yarp-ports/tree/solution).

**Comment by accateo** [comment]
Thanks!
